PixelHunter1995/GameStates/Menu.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Microsoft.Xna.Framework.Input;
+using System.Collections.Generic;
+
+namespace PixelHunter1995.GameStates
+{
+    /// <summary>
+    /// State when being in main menu
+    /// </summary>
+    class Menu : IGameState
+    {
+        private readonly Texture2D menu;
+        private readonly SpriteBatch spriteBatch;
+        private bool escapeHaveBeenUp;
+
+        public Menu(Texture2D menu, SpriteBatch spriteBatch)
+        {
+            this.menu = menu;
+            this.spriteBatch = spriteBatch;
+            this.escapeHaveBeenUp = false;
+        }
+
+        public void Draw(GameTime gameTime)
+        {
+            spriteBatch.Begin();
+            spriteBatch.Draw(menu, Vector2.Zero, Color.White);
+            spriteBatch.End();
+        }
+
+        public IGameState Update(GameTime gameTime, Dictionary<string, IGameState> gameStates)
+        {
+            KeyboardState state = Keyboard.GetState();
+            if (state.IsKeyUp(Keys.Escape))
+            {
+                escapeHaveBeenUp = true;
+            }
+
+            if (state.IsKeyDown(Keys.Escape) && escapeHaveBeenUp)
+            {
+                return gameStates["exit"];
+            }
+            else if (state.IsKeyDown(Keys.Enter))
+            {
+                escapeHaveBeenUp = false;
+                return gameStates["playing"];
```

Comment:
I think it might be possible that we want to create a new state instance here, but I don't know that, so let's keep it like it is (i.e. the states are initialized at game start).